clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcher.java

```diff
@@ -117,11 +104,6 @@ else if (timestamp == ListOffsetsRequest.LATEST_TIMESTAMP)
      *                                                                         and one or more partitions aren't awaiting a seekToBeginning() or seekToEnd().
      */
     public void resetPositionsIfNeeded() {
-        // Raise exception from previous offset fetch if there is one
-        RuntimeException exception = cachedListOffsetsException.getAndSet(null);
```

Comment:
his seems a change in existing logic and not just a refactoring. Is this change expected?